### PR TITLE
Refactor update_backend.yml workflow

### PR DESCRIPTION
**Pull request by sefffal**
@MilesCranmer This PR includes my fixes to this workflow. It should, when run, detect a new backend version, bump the python patch number, and open a PR.

Note that you may need to run modify some repository settings to allow additional action permissions.

### CODE REVIEW

File: .github/workflows/update_backend.yml [hunk 1]
```diff
@@ -13,13 +16,13 @@ jobs:
...
 
-      - name: "Install PySR"
+      - name: "Install pysr"
         run: |
           python -m pip install --upgrade pip
           pip install -r requirements.txt
           python setup.py install
           # Not needed:
-          # python -c 'import pysr; pysr.install()'
+          # python -c 'import octofitterpy; octofitterpy.install()'
```
Comment (MilesCranmer) [inline .github/workflows/update_backend.yml:25]: Can remove these changes

File: .github/workflows/update_backend.yml [hunk 2]
```diff
@@ -28,39 +31,57 @@ jobs:
           git clone https://github.com/MilesCranmer/SymbolicRegression.jl
           cd SymbolicRegression.jl
           echo "version=$(git describe --tags --abbrev=0 | sed 's/^v//')" >> $GITHUB_OUTPUT
+          echo "version=$(git describe --tags --abbrev=0 | sed 's/^v//')"
```
Comment (MilesCranmer) [inline .github/workflows/update_backend.yml:34]: Why was this one changed?

File: .github/workflows/update_backend.yml (hunk 1 above)
Comment (MilesCranmer) [reply .github/workflows/update_backend.yml:25]: (And others like this one)

File: .github/workflows/update_backend.yml (hunk 2 above)
Comment (sefffal) [reply .github/workflows/update_backend.yml:34]: Debugging this workflow with no output pretty hard, so I added echos throughout so the different versions are in the logs. You're welcome to remove them of course.

File: .github/workflows/update_backend.yml (hunk 1 above)
Comment (sefffal) [reply .github/workflows/update_backend.yml]: my bad, fixed.

File: .github/workflows/update_backend.yml (hunk 2 above)
Comment (MilesCranmer) [reply .github/workflows/update_backend.yml:34]: Oh wait, sorry, I see that the version is now stored in `pysr/version.py`, so this part actually isn't needed?

**Review by MilesCranmer — APPROVED** [review]

**Pull request merged**